facade/src/main/java/org/jboss/pnc/facade/providers/ArtifactProviderImpl.java

```diff
@@ -384,6 +385,25 @@ public org.jboss.pnc.dto.Artifact getSpecificFromPurl(String purl) {
         return mapper.toDTO(artifact);
     }
 
+    @Override
+    public Page<org.jboss.pnc.dto.Artifact> getDeliveredArtifactsSharedInMilestones(
```

Comment:
You can leave this method in PMProvider. On facade layer, you can have whatever model's Repository or even use multiple repositories (see BuildProvider). Just the Repository layer should be compliant by the return type, the facade(Providers) layer is more... contextual??

Reply:
this might be problem bcs `queryForCollection` is inherited from AbstractProvider
so in `milestoneProvider`, I would need to add dependency to `artifactProvider`
(milestone's `queryForCollection` wouldn't do for page of Artifacts)

Reply:
in the end i added the artifactProvider dependency, both providers are @Stateless so no circular dependency should happen

Reply:
no sorry, you've had it correct the first time... I got confused, been a long time since adding things in facade.